app/scripts/components/exploration/timeline.tsx

```diff
@@ -0,0 +1,472 @@
+              renderTriggerElement={(props) => (
+                <DatePickerButton {...props} size='small'>
+                  <span className='head-reference'>L</span>
+                  <span>
+                    {selectedInterval
+                      ? format(selectedInterval.start, 'MMM do, yyyy')
+                      : 'Date'}
+                  </span>
+                  <span className='head-reference'>R</span>
+                  <span>
+                    {selectedInterval
+                      ? format(selectedInterval.end, 'MMM do, yyyy')
+                      : 'Date'}
+                  </span>
+                  <CollecticonChevronDownSmall />
+                </DatePickerButton>
+              )}
+            />
+          </ControlsToolbar>
+
+          <DateAxis xScaled={xScaled} width={width} />
+        </TimelineControls>
+      </TimelineHeader>
+      <TimelineContent>
+        {selectedDay ? (
+          <TimelineHeadP
+            domain={dataDomain}
+            xScaled={xScaled}
+            onDayChange={setSelectedDay}
+            selectedDay={selectedDay}
+            width={width}
+          />
+        ) : (
+          false
+        )}
+        {selectedInterval && (
+          <>
+            <TimelineHeadL
+              domain={dataDomain}
+              xScaled={xScaled}
+              onDayChange={(d) => {
+                setSelectedInterval((interval) => {
+                  const prevDay = sub(interval!.end, { days: 1 });
+                  return {
+                    end: interval!.end,
+                    start: isAfter(d, prevDay) ? prevDay : d
+                  };
+                });
+              }}
+              selectedDay={selectedInterval.start}
+              width={width}
+            />
+            <TimelineHeadR
+              domain={dataDomain}
+              xScaled={xScaled}
+              onDayChange={(d) => {
+                setSelectedInterval((interval) => {
+                  const nextDay = add(interval!.start, { days: 1 });
+                  return {
+                    start: interval!.start,
+                    end: isBefore(d, nextDay) ? nextDay : d
+                  };
+                });
+              }}
+              selectedDay={selectedInterval.end}
+              width={width}
+            />
+            <TimelineRangeTrack
+              range={selectedInterval}
+              xScaled={xScaled}
+              width={width}
+            />
+          </>
+        )}
+
+        <DateGrid width={width} xScaled={xScaled} />
+
+        <TimelineContentInner ref={datasetsContainerRef}>
+          <DatasetList
+            datasets={datasets}
+            width={width}
+            xScaled={xScaled}
+            selectedDay={selectedDay}
+          />
+        </TimelineContentInner>
+      </TimelineContent>
+    </TimelineWrapper>
+  );
+}
+
+export default Timeline;
+
+function DatasetList(props: any) {
+  const { datasets, ...rest } = props;
```

Comment:
I would recommend against passing down `datasets`, and directly use the atom value instead:

```suggestion
  const datasets = useAtomValue(timelineDatasetsAtom);
```

IMHO it only makes sense to pass down a prop containing data when the component renders an element of an array, ie `<Rabbit rabbitData={rabbits[i]} />`